client/src/cordova/android/README.md

```diff
@@ -6,17 +6,43 @@ The main entrypoint to Android's Java code is `client/src/cordova/plugin/android
 
 ## Set up your environment
 
-Additional requirements for Android:
+Install these pre-requisites:
 
-- [Java Development Kit (JDK) 11](https://jdk.java.net/archive/)
-  - Set `JAVA_HOME` environment variable if you have multiple JDK versions installed. On macOS: `export JAVA_HOME=$(/usr/libexec/java_home -v 11.0)`.
-- Latest [Android Sdk Commandline Tools](https://developer.android.com/studio/command-line) ([download](https://developer.android.com/studio#command-line-tools-only))
-  - Place it at `$HOME/Android/sdk/cmdline-tools/latest`
-  - Set `ANDROID_HOME` (e.g., `$HOME/Android/sdk`) and `ANDROID_NDK` (e.g., `$ANDROID_HOME/ndk`) environment variables
-- Android SDK 34 (with build-tools) via commandline `$HOME/Android/sdk/cmdline-tools/latest/bin/sdkmanager "platforms;android-34" "build-tools;34.0.0"`
-- [Gradle 7.3+](https://gradle.org/install/). On macOS: `brew install gradle`.
+- [Java Development Kit (JDK) 17+](https://jdk.java.net/archive/). On macOS:
 
-[Android Studio 2020.3.1+](https://developer.android.com/studio) is not required, but it's helpful if you are developing Android code.
+  ```shell
+  brew install openjdk@17
+
+  sudo ln -sfn $HOMEBREW_PREFIX/opt/openjdk@17/libexec/openjdk.jdk /Library/Java/JavaVirtualMachines/openjdk-17.jdk
+
+  export CORDOVA_JAVA_HOME=$(/usr/libexec/java_home -v 17.0)
+  ```
+
+- [Gradle 8.7+](https://gradle.org/install/). On macOS: `brew install gradle`.
+
+Then we need to install and configure the Android components. You can follow the [Cordova Android Platform Guide](https://cordova.apache.org/docs/en/latest/guide/platforms/android/index.html), which uses Android Studio to configure the environment. You need an Android Studio compatible with the Android Gradle Plugin 8.3.0 (see [build.gradle](client/src/cordova/android/OutlineAndroidLib/build.gradle)) we use ([compatibility table](https://developer.android.com/studio/releases#android_gradle_plugin_and_android_studio_compatibility)).
+
+Alternatively, you can do it on the command-line:
+
+1. Set environmental variables:
+
+    ```shell
+    export ANDROID_HOME=$HOME/Android/sdk
```

Comment:
This is the Linux env var, on mac it should be `export ANDROID_HOME=~/Library/Android/sdk`: https://cordova.apache.org/docs/en/latest/guide/platforms/android/index.html#linux-and-macos

Shall we switch to the mac env var since mac is the recommended development environment?

Reply:
This is the variable we have been using on macOS. The one you are suggesting is the one used by Android Studio. I guess we kept them separate to avoid conflict? The Linux one is actually different, with capital S in `Sdk`.

I could test if merging with Android Studio would work, but that's more effort and this is not a change, and I'd rather focus on help Jina right now.

Reply:
Ok, moved to using the same path and Android Studio. Also tested it from scratch and it worked fine

Reply:
I did find an issue: the ndk installation was missing!

Reply:
yes, I guess you still need to install the NDK from Android Studio (or command line). But it should work, on Linux I used the Android Studio path.

Reply:
Apparently the AGP will install the NDK automatically. So after you use the Gradle build, it will be there. But the first time it fails on the tun2socks build.

BTW, I updated the instructions to also install the sources and the image for the emulator.